src/main/java/pl/javastart/couponcalc/PriceCalculator.java

```diff
@@ -5,9 +5,65 @@
 public class PriceCalculator {
 
     public double calculatePrice(List<Product> products, List<Coupon> coupons) {
-        // TODO
-
+        double price;
+        if (products == null) {
+            return 0;
+        } else {
+            for (Product product : products) {
+                if (coupons == null) {
+                    return product.getPrice();
+                } else {
+                    for (Coupon coupon : coupons) {
+                        if (products.size() > 1) {
+                            if (coupons.size() > 1) {
+                                return getSumPriceForFewCoupons(products, coupons);
+                            } else {
+                                return getSumPriceForFewProducts(products, coupons);
+                            }
```

Comment:
Wydaje mi się, że metody `getSumPriceForFewCoupons` i `getSumPriceForFewProducts` robią dokładnie to samo.